### DISCUSSION TITLE
Unable to understand how to run the project in development mode

### DISCUSSION THREAD

**Discussion by AbhinavJuyal** [gpbl/react-day-picker#2478] [Support] [answered]
I cloned this repo, looked into package.json.
I couldn't figure out how to run the repo in development mode.
Do I have to run the `build` command everytime I make some changes and then re-run the `examples-app` project to see the changes? It didn't seem to be an efficient way to do it.

Am I missing something?

**Comment by gpbl** [comment]
Hi @AbhinavJuyal,

Thanks for your feedback. Indeed, we are missing the development docs that were present in the previous version.

If you use Visual Studio Code, open the [react-day-picker.code-workspace](https://github.com/gpbl/react-day-picker/blob/main/react-day-picker.code-workspace) to set up the development environment. This workspace includes:

- Suggested extensions for the project
- TypeScript compiler running in watch mode for the ESM export
- Docusaurus website running at http://localhost:2001, with documentation and API docs
- Vite.js examples app served at http://localhost:5173
  - Use this app to run isolated test cases and examples, exported by `/examples/index.ts`

DayPicker development is test-driven. Change code and run the tests via `pnpm run test-watch` or `pnpm run test`. I personally use the Jest plugin in the VS Code workspace, which works fairly well.

If you are not using Visual Studio Code, here are some useful scripts:

- To run the build in watch mode:
  ```sh
  pnpm tsc --project tsconfig-esm.json --watch
  ```
- To run the tests in watch mode:
  ```
  pnpm run test-watch
  ```
- To run the examples app at http://localhost:5173
  ```
  pnpm --filter examples-app run dev
  ```
- To run the documentation at http://localhost:2001:
  ```
  pnpm --filter website run start
  ```

Looking forward to know more about your experience hacking DayPicker 👀

**Comment by gpbl** [comment, marked as answer]
Please see https://daypicker.dev/development/contributing#working-with-the-source-code for the latest development docs. Thanks!
